src/Nethermind/Nethermind.Blockchain/Synchronization/SyncModeSelector.cs

```diff
@@ -28,13 +28,15 @@ public class SyncModeSelector : ISyncModeSelector
         private readonly ISyncProgressResolver _syncProgressResolver;
         private readonly IEthSyncPeerPool _syncPeerPool;
         private readonly ISyncConfig _syncConfig;
+        private readonly IBlockTree _blockTree;
         private readonly ILogger _logger;
 
-        public SyncModeSelector(ISyncProgressResolver syncProgressResolver, IEthSyncPeerPool syncPeerPool, ISyncConfig syncConfig, ILogManager logManager)
+        public SyncModeSelector(ISyncProgressResolver syncProgressResolver, IEthSyncPeerPool syncPeerPool, ISyncConfig syncConfig, IBlockTree blockTree, ILogManager logManager)
         {
```

Comment:
please do not add blocktree dependency here, sync progress resolcer has this responsibility